### ISSUE TITLE
Jupyterlab deletes all files in linked extension

### ISSUE THREAD

**Issue by vidartf** [jupyterlab/jupyterlab#3212]
Using jupyterlab 0.28.15 from conda, I'm linking in two npm packages from a lerna managed repository (one extension, and one non-extension dependency). When running any subsequent command that causes a jupyterlab re-build, jupyterlab deletes all the files in the directory of the non-extension package. This causes the subsequent build to fail, as it cannot find the package.json file.

Note: This is likely only a Windows issue, as npm/lerna uses junctions for linking packages, which might get treated differently that symlinked directories. See also https://github.com/npm/npm/issues/19091.

**Comment by orpheus92** [comment]
I had similar issues with windows before. Not sure why it was happening.

**Comment by blink1073** [comment]
Yeah, I've seen this on Windows when running `git clean -dfx` in a Lerna repo.

**Comment by vidartf** [comment]
@blink1073 Yes, but I'm not calling git clean myself. I've seen git commands in npm scripts recently, but I *think* I observed this behaviour before that.

**Comment by blink1073** [comment]
I have a workaround for the `git clean -dfx` problem [here](https://github.com/jupyterlab/jupyterlab/pull/3270/commits/19ab13f4532b4d22c1168f17ef0cd7fa8eecc657).)

**Comment by blink1073** [comment]
I completely overhauled the labextension implementation in #3168, and am unable to reproduce the linking bug in master, would you mind trying against master, @vidartf?

**Comment by vidartf** [comment]
I'm trying to test this now, but I'm having troubles with the link step:
```cmd
C:\dev\nbdime\packages\nbdime [jlab-ext ≡ +1 ~1 -0 !]> jupyter labextension link .
> "C:\Program Files\nodejs\npm.CMD" pack .
npm ERR! code ENOLOCAL
npm ERR! Could not install from "" as it does not contain a package.json file
```

**Comment by blink1073** [comment]
Okay, I'll try locally with nbdime as well.  Thanks!

**Comment by blink1073** [comment]
Ah, that error and #3279 are fixed in #3270.

**Comment by vidartf** [comment]
Now that #3270 got merged I tried again. Current output:
```cmd
C:\dev\nbdime\packages\nbdime [config +2 ~1 -0 !]> jupyter labextension link .
> "C:\Program Files\nodejs\npm.CMD" pack C:\dev\nbdime\packages\nbdime
nbdime-0.4.0.tgz
Installing C:\dev\nbdime\packages\nbdime as a linked package:
No `jupyterlab` key
[WinError 145] The directory is not empty: 'C:\\Miniconda3\\share\\jupyter\\lab\\staging\\node_modules\\falafel\\node_modules'
```

Not sure what to do about that falafel.

**Comment by vidartf** [comment]
Clearing staging seems to have done the trick. 🎉

**Comment by vidartf** [comment]
Can no longer reproduce on master, closing as solved by #3270.
